common/hooks/pre-pkg/03-rewrite-python-shebang.sh

```diff
@@ -13,15 +13,28 @@ hook() {
 	fi
 
 	if [ -n "$pyver" ]; then
-		shebang="#!/usr/bin/python${pyver%.*}"
+		default_shebang="#!/usr/bin/python${pyver%.*}"
 	fi
 
 	find "${PKGDESTDIR}" -type f -print0 | \
 		while IFS= read -r -d '' file; do
 			[ ! -s "$file" ] && continue
-			[ -z "$(sed -n -E -e 2q -e '/^#!.*([[:space:]]|\/)python([0-9]\.[0-9])?([[:space:]]+|$)/p' "$file")" ] && continue
-			[ -n "$shebang" ] || msg_error "cannot convert shebang, set python_version\n"
-			echo "   Shebang converted to '$shebang': ${file#$PKGDESTDIR}"
+
+			pyinterp=$(sed -n -E -e 2q -e 's@^#!.*([[:space:]]|/)(python([0-9][0-9.]*)?)([[:space:]]+|$)@\2@p' "$file")
```

Comment:
With the change in https://github.com/void-linux/void-packages/pull/22698#discussion_r437456942

a.k.a using `shebang="#!/usr/bin/python${pyver%%.*}"`, we can simplify as my original suggestion.

`python...` is problematic, we may just treat it as `python`.

Thoughts?

Reply:
I'm on the fence. In practice, things like `python...` or `python.2` will probably never appear and it won't matter. But if some interpreter like `python.2` is ever listed in a shebang, the question is whether we should ignore that file because it doesn't look like the "typical" python, force it to the default python shebang, or force it to an unversioned `/usr/bin/python`. 

The third option would be the behavior of the current implementation if we just change the pattern to `(python[0-9.]*)` because `pyver` would be `.<something>` and `${pyver%%.*}` would be empty. This is probably undesirable because, if we allow unversioned shebangs here, why do we not allow them everywhere? Mapping `python.2` to `python` but `python` to `python3` or `python2` violates the principle of least surprise.

The second option doesn't seem quite right either, because we could end up forcing something like `python.2` to `python3` depending on automatic module version detection or the value of `python_version`.

The first option, ignoring these shebangs, at least takes a conservative approach that we won't change things that don't look recognizable.